cloud/deployment/inspect/inspect.go

```diff
@@ -145,136 +155,168 @@ func Inspect(wsID, deploymentName, deploymentID, outputFormat string, client ast
+		if coreDeployment.TaskPodNodePoolId != nil {
+			defaultWorkerType, err = GetDefaultWorkerType(*coreDeployment.TaskPodNodePoolId, *coreDeployment.ClusterId, platformCoreClient)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	return map[string]interface{}{
-		"name":                 sourceDeployment.Label,
-		"description":          sourceDeployment.Description,
-		"workspace_name":       sourceDeployment.Workspace.Label,
-		"deployment_type":      sourceDeployment.Type,
-		"cloud_provider":       sourceDeployment.Cluster.CloudProvider,
-		"region":               sourceDeployment.Cluster.Region,
+
+	deploymentMap := map[string]interface{}{
+		"name":                 coreDeployment.Name,
+		"description":          coreDeployment.Description,
+		"workspace_name":       coreDeployment.WorkspaceName,
+		"deployment_type":      string(*coreDeployment.Type),
+		"cloud_provider":       coreDeployment.CloudProvider,
+		"region":               coreDeployment.Region,
 		"cluster_name":         clusterName,
-		"runtime_version":      sourceDeployment.RuntimeRelease.Version,
-		"dag_deploy_enabled":   sourceDeployment.DagDeployEnabled,
-		"ci_cd_enforcement":    sourceDeployment.APIKeyOnlyDeployments,
-		"scheduler_size":       sourceDeployment.SchedulerSize,
-		"is_high_availability": sourceDeployment.IsHighAvailability,
-		"scheduler_au":         sourceDeployment.DeploymentSpec.Scheduler.AU,
-		"scheduler_count":      sourceDeployment.DeploymentSpec.Scheduler.Replicas,
-		"executor":             sourceDeployment.DeploymentSpec.Executor,
+		"runtime_version":      coreDeployment.RuntimeVersion,
+		"dag_deploy_enabled":   coreDeployment.DagDeployEnabled,
+		"ci_cd_enforcement":    coreDeployment.IsCicdEnforced,
+		"is_high_availability": coreDeployment.IsHighAvailability,
+		"scheduler_au":         coreDeployment.SchedulerAu,
+		"scheduler_count":      coreDeployment.SchedulerReplicas,
+		"executor":             coreDeployment.Executor,
+		"scheduler_size":       coreDeployment.SchedulerSize,
 	}
+	if deployment.IsDeploymentStandard(*coreDeployment.Type) || deployment.IsDeploymentDedicated(*coreDeployment.Type) {
+		deploymentMap["default_task_pod_cpu"] = coreDeployment.DefaultTaskPodCpu
+		deploymentMap["default_task_pod_memory"] = coreDeployment.DefaultTaskPodMemory
+		deploymentMap["resource_quota_cpu"] = coreDeployment.ResourceQuotaCpu
+		deploymentMap["resource_quota_memory"] = coreDeployment.ResourceQuotaMemory
+	}
+	if !deployment.IsDeploymentStandard(*coreDeployment.Type) {
+		deploymentMap["default_worker_type"] = defaultWorkerType
+	}
+
+	return deploymentMap, nil
 }
 
-func getAdditional(sourceDeployment *astro.Deployment) map[string]interface{} {
-	qList := getQMap(sourceDeployment.WorkerQueues, sourceDeployment.Cluster.NodePools, sourceDeployment.DeploymentSpec.Executor, sourceDeployment.Type)
+func getAdditional(coreDeployment *astroplatformcore.Deployment, nodePools []astroplatformcore.NodePool) map[string]interface{} {
```

Comment:
Can we make the function name more explicit
```suggestion
func getAdditionalNullableFields(coreDeployment *astroplatformcore.Deployment, nodePools []astroplatformcore.NodePool) map[string]interface{} {
```

Reply:
updated